tests/components/apcupsd/test_config_flow.py

```diff
@@ -0,0 +1,216 @@
+
+
+async def test_flow_minimal_status(hass: HomeAssistant) -> None:
+    """Test successful creation of config entries via user configuration when minimal status is reported."""
+    with patch("apcaccess.status.parse") as mock_parse, patch(
+        "apcaccess.status.get", return_value=b""
+    ), _patch_setup() as mock_setup:
+        # We test different combinations of minimal statuses, where the title of the
+        # integration will vary.
+        extras = (
+            {"UPSNAME": "Friendly Name"},
+            {"MODEL": "MODEL X"},
+            {"SERIALNO": "ZZZZ"},
+            {},
+        )
+        for i, extra in enumerate(extras):
+            status = MOCK_MINIMAL_STATUS | extra
+            mock_parse.return_value = status
+
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={CONF_SOURCE: SOURCE_USER}, data=CONF_DATA
+            )
+            await hass.async_block_till_done()
+            assert result["type"] == FlowResultType.CREATE_ENTRY
+
+            assert result["data"] == CONF_DATA
+
+            assert mock_setup.call_count == i + 1
+
+            # Clear the config entry for next iteration.
+            for entry in hass.config_entries.async_entries(DOMAIN):
+                await hass.config_entries.async_remove(entry.entry_id)
+            if "UPSNAME" in status:
+                assert result["title"] == status["UPSNAME"]
+            elif "MODEL" in status:
+                assert result["title"] == status["MODEL"]
+            elif "SERIALNO" in status:
+                assert result["title"] == status["SERIALNO"]
+            else:
+                assert result["title"] == "APC UPS"
+
+            # Clear the config entry for next iteration.
+            for entry in hass.config_entries.async_entries(DOMAIN):
+                await hass.config_entries.async_remove(entry.entry_id)
+
+            assert mock_setup.call_count == i + 1
+
+
+async def test_flow_import(hass: HomeAssistant) -> None:
+    """Test successful creation of config entries via YAML import."""
+    with patch("apcaccess.status.parse", return_value=MOCK_STATUS), patch(
+        "apcaccess.status.get", return_value=b""
+    ), _patch_setup() as mock_setup:
+        # Importing from YAML will create an extra field CONF_RESOURCES in the config
+        # entry, here we test if it is properly stored.
+        resources = ["MODEL"]
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=CONF_DATA | {CONF_RESOURCES: resources},
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == FlowResultType.CREATE_ENTRY
+        assert result["title"] == MOCK_STATUS["MODEL"]
+        assert result["data"] == CONF_DATA | {CONF_RESOURCES: resources}
+
+        mock_setup.assert_called_once()
+
+
+async def test_flow_import_with_invalid_resource(hass: HomeAssistant) -> None:
```

Comment:
I don't think we check any resources in the config flow.

Reply:
Right, this test is actually testing the integration instead of config flow. 

Removed, and will add this back in a separate PR.